src/OpenToolkit.Mathematics/Geometry/Box2.cs

```diff
@@ -121,6 +121,160 @@ public Vector2 Center
+        {
+            get => _max.Y - _min.Y;
+            set => _max.Y = _min.Y + value;
+        }
+
+        public float Left
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Top
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float Right
+        {
+            get => _max.X;
+            set => _max.X = value;
+        }
+
+        public float Bottom
+        {
+            get => _max.Y;
+            set => _max.Y = value;
+        }
+
+        public float X
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Y
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float SizeX
+        {
+            get => _max.X - _min.X;
+        }
+
+        public float SizeY
+        {
+            get => _max.Y - _min.Y;
+        }
+
+        public Vector2 Location => _min;
+
+        public bool IsEmpty => _min.X == 0 && _min.Y == 0 && _max.X == 0 && _max.Y == 0;
+
+        public static readonly Box2 Zero = new Box2(0, 0, 0, 0);
+
+        public static readonly Box2 Identity = new Box2(0, 0, 1, 1);
+
+        public static Box2 FromSize(Vector2 location, Vector2 size)
+        {
+            return new Box2(location, location + size);
+        }
+
+        public static Box2 FromPositions(Vector2 min, Vector2 max)
+        {
+            return new Box2(min, max);
+        }
+
+        public static Box2 FromPositions(float minX, float minY, float maxX, float maxY)
+        {
+            return new Box2(minX, minY, maxX, maxY);
+        }
+
+        public void Intersect(Box2 other)
+        {
+            Box2 result = Box2.Intersect(other, this);
+
+            X = result.X;
+            Y = result.Y;
+            Width = result.Width;
+            Height = result.Height;
+        }
+
+        public static Box2 Intersect(Box2 a, Box2 b)
+        {
+            float minX = a._min.X > b._min.X ? a._min.X : b._min.X;
+            float minY = a._min.Y > b._min.Y ? a._min.Y : b._min.Y;
+            float maxX = a._max.X < b._max.X ? a._max.X : b._max.X;
+            float maxY = a._max.Y < b._max.Y ? a._max.Y : b._max.Y;
+
+            if (maxX >= minX && maxY >= minY)
+            {
+                return new Box2(minX, minY, maxX, maxY);
+            }
+            return Box2.Zero;
+        }
+
+        public bool IntersectsWith(Box2 other)
+        {
+            return other._min.X < _max.X
+                && _min.X < other._max.X
+                && other._min.Y < _max.Y
+                && _min.Y < other._max.Y;
+        }
+
+        public bool TouchWith(Box2 other)
+        {
+            return other._min.X <= _max.X
+                && _min.X <= other._max.X
+                && other._min.Y <= _max.Y
+                && _min.Y <= other._max.Y;
+        }
+
+        public static Box2 Union(Box2 a, Box2 b)
```

Comment:
I would also like to see a `Box2 Inflate(Vector2 pos)` which extends the box to include the point if it's not already in the box.

I know `Union` is the correct name for this but personally I would have named this `Inflate` to match behaviors between the two functions (they both make the box bigger). But that's just my preference, both naming schemes work.

Reply:
We already have a `void Inflate(Vector2 point)` and `Box2 Inflated(Vector2 point)`, so that's not part of my patch.

Regarding `Union (Box2 a, Box2 b)`: I prefer `Union` but if somebody else thinks `Inflate` is the better choice, we can change it.

Reply:
I missed the fact that there existed an `Inflate(...)` already...
I'm fine with both names, I was just expressing my preference.

Reply:
Well a `Union` in my head would be more of a:
```
UnionBox2D
{
 List<Box2D> boxes;
 bool Intersect(Vector2 point) => boxes.Any(x => x.Intersect(point));
}
```
Therefore I think inflate still matches better.